### ISSUE TITLE
Node core now supports similar functionality - add a note about it?

### ISSUE THREAD

**Issue by Tyler-Murphy** [JsCommunity/async-iterator-to-stream#1]
It was added in node v12.3: https://nodejs.org/api/stream.html#stream_stream_readable_from_iterable_options.

I've used this library, but I just discovered that node now has a helper to do the same thing. It seems the only major difference is that you can't pass a generator function directly.

I think it'd be helpful to add a note to the readme to point people to the built-in solution, and explain the differences so that people can decide which to use.

**Comment by julien-f** [comment]
Indeed, it's very different, the ability to create a stream with a generator is the main goal of this library, for instance, I'm using it to [generate a synthetic VHD](https://github.com/vatesfr/xen-orchestra/blob/master/packages/vhd-lib/src/createSyntheticStream.js) file from a chain of delta VHD files, which would, IMHO be difficult to do with the stream API, even more when correctly handling back-pressure.

Anyway, I agree with that a note would be nice, feel free to open a PR :slightly_smiling_face:

**Comment by Tyler-Murphy** [comment]
I may be wrong, since I haven't used this feature, but I think `async-iterator-to-stream` supports this pattern, where a generator function can be bound, as in the `createReadStream` in the readme:

```js
const makeCountStream = asyncIteratorToStream(function * count () {
  yield 1
  yield 2
  ...
})

makeCountStream().pipe(process.stdout)
```

`stream.Readable.from` only accepts iterables, not generators. Meaning that to accomplish the same thing you'd have to do this:

```js
async function * count () {
  yield 1
  yield 2
  ...
}

const makeCountStream = () => stream.Readable.from(count())

makeCountStream().pipe(process.stdout)
```

I don't think this is supported:

```js
stream.Readable.from(function * doSomething () {
  yield 1
  yield 2
  ...
})
```

Either way, it seems like a minor difference. I'll open a PR!

**Comment by julien-f** [comment]
AFAIK the differences are:

- ability to create stream factory from (async) generators
- support Node >= 6
- support sync generators yielding promises instead of async generators

I think, as soon as the new `Readable.from` is included in [readable-stream](https://github.com/nodejs/readable-stream/), I will migrate to using this for recent versions of Node.
